src/drt/src/db/obj/frVia.h

```diff
@@ -193,7 +193,8 @@ class frVia : public frRef
    */
   bool hasPin() const override
   {
-    return (owner_) && (owner_->typeId() == frcPin);
+    return (owner_) && ((owner_->typeId() == frcBPin) ||
+                         owner_->typeId() == frcMPin);
```

Comment:
I think a better solution is to move this method to `frShape` because all of its children have this function. Then it will only be in `frVia` and `frShape`.